extracted elasticsearch-logging service name as environment variable

mrahbar opened the pull request:
**What this PR does / why we need it**:
Deploying the cluster-addon fluentd-elasticsearch with customized resource definitions can cause elasticsearch discovery to fail because the service name `elasticsearch-logging` is hard-coded in  cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
-> none yet

**Special notes for your reviewer**:
The name of the  environment variable is ELASTICSEARCH_SERVICE_NAME. When non is given the fallback service-name  fallback is  `elasticsearch-logging`

```release-note
[fluentd-elasticsearch addon] Elasticsearch service name can be overridden via env variable ELASTICSEARCH_SERVICE_NAME
```

crassirostris commented:
SGTM

Have you built a test image and tried it in a cluster?

mrahbar commented:
@crassirostris I'm currently on it but I can't notice any drawbacks

crassirostris commented:
Sure, I'm not talking about any drawbacks, I just want to verify that everything builds and runs as expected :)

/ok-to-test

mrahbar commented:
I build it and it works as expected :)
I also signed the cla.

crassirostris commented:
/retest

mrahbar commented:
Any updates on this PR? The cause of the failing tests seems unrelated.

crassirostris commented:
I'll take a look by the EOW

crassirostris commented:
@mrahbar OK, seems fine. Could you please do the following?

- update the image version in the `Makefile`
- write the release note (include `[fluentd-elasticsearch plugin]` there)

mrahbar commented:
I change the image version on Makefile from v5.6.2 to v5.6.3
Where do I supposed to write the release notes?

crassirostris commented:
Sorry, forgot about it. This version we try to correlate with the version of the Elasticsearch, e.g. currently the version of Elasticsearch and of the image is `5.6.2`. Could you please replace `5.6.3` with `5.6.2-1`?

> Where do I supposed to write the release notes?

In the PR description, in the following format:

````
```release-note
[fluentd-elasticsearch addon] ...
```
````

mrahbar commented:
Any updates on this?
@crassirostris do you know why the summary says "commit missing GitHub user"? I added my commit email address to GitHub

crassirostris commented:
@mrahbar Sorry, it slipped past me

Regarding `commit missing GitHub user`, two out of three your commits are marked as not associated with your GitHub account. Maybe you pushed them from a different computer, with a different email setup in git config?

Overall SGTM, last comments:

- Make release note more crisp, something like `Elasticsearch service name now can be configured via env variable`
- Squash commits into one
- Use the new Elasticsearch version in the DaemonSet definition or make a follow-up PR to do this

mrahbar commented:
- I changed the release note as you proposed
- I squashed the commits and hopefully added all my email adresses to my account for cla to work
- Regarding: 'Use the new Elasticsearch version in the DaemonSet ' I would prefer creating a new PR because it is unrelated to this PR

crassirostris commented:
@mrahbar Thanks!

> I changed the release note as you proposed

My main point was to leave out the specific details. Release notes consist of hundreds of messages, technical details only makes them harder to read. If somebody wants the details, this person will open the PR or docs

> I would prefer creating a new PR because it is unrelated to this PR

Sounds good. Please assign this PR to me

mrahbar commented:
@crassirostris I created a new PR  #55400 for version update of elasticsearch and kibana but I can't assign it to you.

mrahbar commented:
Update of elasticsearch and kibana version is now in PR #55400

crassirostris commented:
@mrahbar Sounds good, thanks

Do you agree with the release note change? It's the last thing, everything else SGTM

crassirostris commented:
@mrahbar Hm, if you going to update the version anyway, just leave out the image version change in this PR, so it will be picked up by the next one

crassirostris commented:
@mrahbar 

Regarding changing the image version. I would really like to avoid creating image releases that are not used, it's a manual process for now

Regarding the notes. I suggest to hear third opinion. @coffeepac could you please take a look?

coffeepac commented:
terse is better for release notes.  I suggest:

```
[fluentd-elasticsearch addon] Elasticsearch service name can be overridden via env variable ELASTICSEARCH_SERVICE_NAME
```

That okay @mrahbar ?  

Also, thanks for the PR!

crassirostris commented:
That also LGTM

mrahbar commented:
Release note also LGTM. Changed it accordingly.

crassirostris commented:
What about image version?

mrahbar commented:
I guess I miss understood you regarding the image version. Resulting in the PR #55400 thus I closed it. I thought you wanted to use this product to also update to the newest es version.  But I guess you just wanted the es image in the daemon-set to be v5.6.2-1. 
I will push the changes shortly.

crassirostris commented:
No-no, sorry for misunderstanding. I suggest to leave the version in this PR unchanged

mrahbar commented:
@crassirostris So then this PR has come to an end. Do you agree. The version update of es will happen in #55400.

crassirostris commented:
@mrahbar No :) I suggest to change the code in this PR and then change the version in another PR. Does it sound reasonable?

mrahbar commented:
@crassirostris can this PR be merge now?

crassirostris commented:
@mrahbar It still includes the Makefile change. Revert this one line (`1.6.2 -> 1.6.2-1`) in this PR and I'll lgtm both of them

mrahbar commented:
@crassirostris I reverted to line in Makefile. I'm currently on my way and can't squash the commits. Is this ok?

crassirostris commented:
@mrahbar Yes, perfect, thanks! LGTM, I'll apply the label right after you squash commits

mrahbar committed:
extracted elasticsearch-logging service name as environment variable ELASTICSEARCH_SERVICE_NAME with fallback on default

mrahbar commented:
@crassirostris Squashed them 👍

crassirostris commented:
/lgtm

coffeepac commented:
/approve no-issue

crassirostris commented:
@coffeepac Thanks!
